### PR TITLE
Issue #46: Add check for non static usages of regex

### PR DESCRIPTION
**Pull request by stil4m**

### CODE REVIEW

**Comment by stil4m** [comment]
@felixLam

File: src/Analyser/Checks/NonStaticRegex.elm [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        let
+            after =
+                f { context | staticEnvironment = False }
+        in
+            { after | staticEnvironment = context.staticEnvironment }
+    else
+        f context
+
+
+isStatic : Function -> Bool
```
Comment (felixLam) [inline src/Analyser/Checks/NonStaticRegex.elm]: maybe rename to isNonStatic to reflect the question that this module answers

File: src/Analyser/Checks/NonStaticRegex.elm [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+onExpressionFunctionReference ( range, inner ) context =
+    case inner of
+        FunctionOrValue "regex" ->
+            addUsedRegex range context
+
+        _ ->
+            context
+
+
+buildRegexImportInformation : List Import -> Maybe FunctionReference
```
Comment (felixLam) [inline src/Analyser/Checks/NonStaticRegex.elm]: This signature sounds like code that is probably implemented elsewhere already, if that is the case maybe that can be extracted into a Util function and reused with more comments.

File: src/Analyser/Checks/NonStaticRegex.elm [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    let
+        regexImport =
+            buildRegexImportInformation fileContext.ast.imports
+    in
+        case regexImport of
+            Nothing ->
+                []
+
+            Just regexImport ->
+                Inspector.inspect
```
Comment (felixLam) [inline src/Analyser/Checks/NonStaticRegex.elm]: While I am not too familiar with `Inspector` the following lines could use a bit more structure and comments.

File: src/Analyser/Checks/NonStaticRegex.elm (hunk 1 above)
Comment (stil4m) [inline src/Analyser/Checks/NonStaticRegex.elm]: I agree, it is now used as `not (isStatic x)` which does the job in my opinion. This function will be lifted out to be used in more checks, but I'm not sure where to yet.

**Pull request merged**